### PR TITLE
chore: improve unit test coverage in @asgardeo/javascript SDK

### PR DESCRIPTION
**Pull request by PasanMadhuranga**
### Purpose

This PR improves unit test coverage for the **@asgardeo/javascript SDK**, addressing the gaps identified in [issue #154](https://github.com/asgardeo/javascript/issues/154).

### Summary of Work Done

* **Fixed and stabilized existing failing tests.**
* **Refined and expanded unit tests** in:

  * `src/api/__tests__/`
  * `src/errors/__tests__/`
* Added coverage for additional code paths, edge cases, and error handling scenarios to improve reliability and regression detection.
* Conducted a manual verification round to ensure all tests pass consistently.

#### Bug Fixes Implemented During Testing

While increasing coverage, a few logic issues were identified and fixed to align with expected SDK behavior.
The following commits address these issues:

| Commit    | Message                                                                                                         |
| --------- | --------------------------------------------------------------------------------------------------------------- |
| [`113a180`](https://github.com/asgardeo/javascript/commit/113a18063a68dc10088a5df02b55f26d3ecb20c0) | `chore(javascript): use numeric ordering for log levels instead of string comparison`                           |
| [`539acf3`](https://github.com/asgardeo/javascript/commit/539acf3df6ed6ea0c94449bdf2eda07e7e2d20e2) | `fix(javascript): handle network and parsing errors gracefully in getUserInfo`                                  |
| [`446de46`](https://github.com/asgardeo/javascript/commit/446de462e95d2bb6ea3f3beda6293519b6909445) | `fix(javascript): improve error handling in ExecuteEmbeddedSignInFlow for invalid URLs and network failures`    |
| [`0c7ec47`](https://github.com/asgardeo/javascript/commit/0c7ec47a1d371c0b9db7ce6a97d98a0815484271) | `fix(javascript): improve error handling in executeEmbeddedSignUpFlow for invalid URLs and network failures`    |
| [`f0e4d8b`](https://github.com/asgardeo/javascript/commit/f0e4d8b71bab6d9702f9398a0f227c1b6485e7a2) | `fix(javascript): improve error handling in initializeEmbeddedSignInFlow for invalid URLs and network failures` |

---

### Observations and Potential Issues

During implementing tests, several **inconsistencies and potential bugs** were noted that may need team clarification or adjustment:

#### 1. `updateMeProfile.ts`

**Issue:**
`requestConfig` can override the explicitly defined HTTP method.

```ts
const requestInit: RequestInit = {
  method: 'PATCH',
  ...requestConfig, // Can override method
  headers: {
    ...requestConfig.headers,
    'Content-Type': 'application/scim+json',
    Accept: 'application/json',
  },
  body: JSON.stringify(data),
};
```

**Suggestion:**
Spread `requestConfig` **after** defining headers and method to ensure the HTTP method cannot be unintentionally overridden.

---

#### 2. Multiple API files

**Files:**
`createOrganization.ts`, `executeEmbeddedSignInFlow.ts`, `executeEmbeddedSignUpFlow.ts`, `getBrandingPreference.ts`,
`getMeOrganizations.ts`, `getOrganization.ts`, `getSchemas.ts`, `getScim2Me.ts`, `getUserInfo.ts`, `updateOrganization.ts`

**Issue:**
`Content-Type` and `Accept` headers can be overridden by `requestConfig.headers`.

```ts
headers: {
  'Content-Type': 'application/json',
  Accept: 'application/json',
  ...requestConfig.headers, // Overrides defaults
},
```

**Suggestion:**
Reverse the merge order if the intention is to enforce standard headers:

```ts
headers: {
  ...requestConfig.headers,
  'Content-Type': 'application/json',
  Accept: 'application/json',
},
```

---

#### 3. `getAllOrganizations.ts`, `initializeEmbeddedSignInFlow.ts`, `updateMeProfile.ts`

**Issue:**
Opposite header precedence. Here, `requestConfig.headers` **is** overridden by `Content-Type` and `Accept`.
This creates inconsistent behavior across API utilities.

---

#### 4. `AsgardeoAPIError.ts`

**Issue:**
`toString()` renders `"undefined"` for `statusText` when not provided.

```ts
public override toString(): string {
    const status = this.statusCode ? ` (HTTP ${this.statusCode} - ${this.statusText})` : '';
    return `[${this.name}] (code="${this.code}")${status}\nMessage: ${this.message}`;
  }
```

---

### Next Steps

* Continue improving test coverage for other under-tested modules such as `utils`.
* Seek confirmation from the team regarding the  **inconsistencies and potential bugs** identified during testing.

---

### Related Issues

* [https://github.com/asgardeo/javascript/issues/154](https://github.com/asgardeo/javascript/issues/154)

### Related PRs

* N/A

---

### Checklist
- [x] Followed the [CONTRIBUTING](https://github.com/asgardeo/javascript/blob/main/CONTRIBUTING.md) guidelines.
- [x] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
- [x] Unit tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by brionmario** [comment]
Hey @PasanMadhuranga,
Great work. I've added few comments.

Also, there's an error in the `@asgardeo/javascript` package test run. Could you please check this.

**Comment by PasanMadhuranga** [comment]
> Shall we fix the test suite errors. Also, lets add a changeset: https://github.com/asgardeo/javascript/blob/main/CONTRIBUTING.md#creating-a-changeset
> 

Hi @brionmario ! I’ve fixed the test suite errors and also added a changeset.

File: packages/javascript/src/api/executeEmbeddedSignInFlow.ts [hunk 1]
```diff
@@ -26,6 +26,18 @@ const executeEmbeddedSignInFlow = async ({
   payload,
   ...requestConfig
 }: EmbeddedFlowExecuteRequestConfig): Promise<EmbeddedSignInFlowHandleResponse> => {
+  try {
```
[the remark this answers is not in the document]
Comment (brionmario) [reply packages/javascript/src/api/executeEmbeddedSignInFlow.ts]: Great!
